### ISSUE TITLE
Weird behavior of tf.train.Saver

### ISSUE THREAD

**Issue by mklopp** [tensorflow/tensorflow#10755]
### System information
- Linux Ubuntu 14.04
- TensorFlow installed from binary:
- TensorFlow version v1.0.0-rc2:
- CUDA: 8.0, CuDNN: 5.1
- Tesla K80, 12GB

### Describe the problem
I have a problem with the tf.train.Saver, specifically with the 'max_to_keep' argument. If I create a Saver with 'max_to_keep' set to let's say 3 and use the saver.save function to save my model in the current directory it keeps all files and doesn't delete the old ones after 3 or more are created. If I set the path where to save the model to a different location it works just fine.

See also my [stackoverflow question](https://stackoverflow.com/questions/44458947/tensorflow-keeps-all-files-how-to-prevent-that)

### Source code / logs
Creates for the numbers 1 to 10 each 3 files:

- testfile-1.data-00000-of-00001
- testfile-1.index
- testfile-1.meta

```
import tensorflow as tf

a = tf.Variable(name='a', initial_value=0)
addops = a+1

saver = tf.train.Saver(max_to_keep=3)
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
sess = tf.Session(config=config)
sess.run(tf.global_variables_initializer())
for i in range(10):
    sess.run(addops)
    save_path = saver.save(sess, 'testfile', global_step=i+1)

sess.close()
```

This code works as expected and deletes all the old files and I only end up with the numbers 8 to 10:

```
import tensorflow as tf

a = tf.Variable(name='a', initial_value=0)
addops = a+1

saver = tf.train.Saver(max_to_keep=3)
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
sess = tf.Session(config=config)
sess.run(tf.global_variables_initializer())
for i in range(10):
    sess.run(addops)
    save_path = saver.save(sess, 'test/testfile', global_step=i+1)

sess.close()
```

**Comment by aselle** [comment]
Could you take a look @concretevitamin , please?

**Comment by concretevitamin** [comment]
Seems relevant to I/O layer.  Adding @rohan100jain.

**Comment by notimesea** [comment]
There is another bug if path to save contains double slash e.g. 'logdir//model'. In this case old checkpoints are not removed as well.

**Comment by rohan100jain** [comment]
Quite sure that https://github.com/tensorflow/tensorflow/commit/3e6c638727c3274908a7c9c6bbf4474c014511fe would have fixed the original issue... If you update your tf version to the latest one, it works.. 

The // case is a little more complex since in some file systems paths have //'s in them... so I'd recommend just not having them in the path :)

**Comment by notimesea** [comment]
@rohan100jain python 'open' works ok with double slash, for me it's expected to have same behaviour with default bash, python semantics of path.

**Comment by adrienycart** [comment]
I have experienced the same problem when using brackets in the name of the saving folder.

**Comment by mklopp** [comment]
Sorry for not following up on this but the original issue has been resolved. So I guess this can be closed. The best way to use the function is to hand over `os.path.abspath(filename)`.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.
